pip/pip-278.md

```diff
@@ -0,0 +1,131 @@
+# Background knowledge
+
+Apache Pulsar is a distributed messaging system that supports multiple messaging protocols and storage methods. Among them, Pulsar Topic Compaction is a mechanism to clean up duplicate messages in topics to reduce storage space and improve system efficiency.
+More topic compaction details can be found in [Pulsar Topic Compaction](https://pulsar.apache.org/docs/en/concepts-topic-compaction/).
+
+# Motivation
+
+Currently, the implementation of Pulsar Topic Compaction is fixed and does not support custom strategy, which limits users from using more Compactor policies in their applications.
+
+For example, we need to parse the Kafka format then compact message in Kop, but the current implementation of Pulsar topic compaction does not support this feature.
```

Comment:
We should explain why Pulsar topic compaction can't work with KoP. Actually, it can work with pulsar format data in KoP, but it can't work with Kafka format data since the data written to the entry is in Kafka format. The Pulsar compactor doesn't aware of Kafka format data. And it doesn't make sense to support Kafka format data handling in Pulsar.

Reply:
Yesh, I add some description to explain why Pulsar topic compaction can't work with KoP.